From 3584983f2548379645ae6e8a2fe62c233f906799 Mon Sep 17 00:00:00 2001
From: Michael W <baddaemon87@gmail.com>
Date: Sat, 6 Jan 2018 13:47:59 +0100
Subject: [PATCH 059/249] wiki: Validate a few properties

* Add validation for cpu_cores, cpu_freq, current_branch, image,
  install_method, kernel, maintainer, name, versions and make all of them
  required fields
* Add validation for custom_unlock_cmd, required_bootloader
  -> optional fields
* Fixup the issues due to wrong formatted fields
* Enforce a codename minimum length of 1
  (Anything else would exclude OnePlus devices ['1', '2', 'X' ...])

Change-Id: I943241062f571d7e5646c16689af6f0b39348329
---
 _data/devices/FP2.yml         |  1 -
 _data/devices/Z00D.yml        |  2 +-
 _data/devices/addison.yml     |  2 +-
 _data/devices/d2att.yml       |  2 +-
 _data/devices/d2spr.yml       |  2 +-
 _data/devices/d2tmo.yml       |  2 +-
 _data/devices/d2vzw.yml       |  2 +-
 _data/devices/ghost.yml       |  2 +-
 _data/devices/gts210ltexx.yml |  2 +-
 _data/devices/gts210wifi.yml  |  2 +-
 _data/devices/ham.yml         |  2 +-
 _data/devices/i9305.yml       |  4 +-
 _data/devices/libra.yml       |  1 -
 _data/devices/m7.yml          |  2 +-
 _data/devices/m7vzw.yml       |  2 +-
 _data/devices/maguro.yml      |  2 +-
 _data/devices/mt2.yml         |  4 +-
 _data/devices/n5100.yml       |  2 +-
 _data/devices/n5110.yml       |  2 +-
 _data/devices/n5120.yml       |  2 +-
 _data/devices/nicki.yml       |  2 +-
 _data/devices/peach.yml       |  2 +-
 _data/devices/t6.yml          |  2 +-
 _data/devices/t6vzw.yml       |  2 +-
 _data/devices/targa.yml       |  2 +-
 _data/devices/wt88047.yml     |  2 +-
 device_sample/sample.yml      |  2 +-
 test/schema-06.yml            | 79 +++++++++++++++++++++++++++++++++++
 28 files changed, 106 insertions(+), 29 deletions(-)

diff --git a/_data/devices/FP2.yml b/_data/devices/FP2.yml
index 6d9acb4..b6bebba 100644
--- a/_data/devices/FP2.yml
+++ b/_data/devices/FP2.yml
@@ -11,7 +11,6 @@ cpu: Krait 400
 cpu_cores: '4'
 cpu_freq: 2.26 GHz
 current_branch: 14.1
-custom_unlock_cmd: ''
 depth: 11 mm (0.43 in)
 download_boot: With the device powered off, hold <kbd>Volume Down</kbd> + <kbd>Power</kbd>. Release when boot logo appears.
 gpu: Adreno 330
diff --git a/_data/devices/Z00D.yml b/_data/devices/Z00D.yml
index 2c0de38..21f2969 100644
--- a/_data/devices/Z00D.yml
+++ b/_data/devices/Z00D.yml
@@ -12,7 +12,7 @@ cpu_cores: '2+2'
 cpu_freq: 1.6 GHz
 current_branch: 14.1
 download_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
-gpu: <kbd>Power</kbd>VR SGX544MP2
+gpu: PowerVR SGX544MP2
 has_recovery_partition: true
 image: Z00A.png
 install_method: fastboot_zenfone
diff --git a/_data/devices/addison.yml b/_data/devices/addison.yml
index 01120ca..74f3d69 100644
--- a/_data/devices/addison.yml
+++ b/_data/devices/addison.yml
@@ -20,7 +20,7 @@ height: 156.4 mm (6.16 in)
 image: addison.png
 install_method: fastboot_motorola
 kernel: android_kernel_motorola_msm8953
-maintainers:
+maintainers: []
 models: [XT1635-02]
 name: Moto Z Play
 network:
diff --git a/_data/devices/d2att.yml b/_data/devices/d2att.yml
index 3383983..e42f302 100644
--- a/_data/devices/d2att.yml
+++ b/_data/devices/d2att.yml
@@ -19,7 +19,7 @@ height:
 image: i9300.png
 install_method: heimdall
 kernel: android_kernel_samsung_d2
-maintainers:
+maintainers: []
 models: [SGH-I747, SGH-I747M]
 name: Galaxy S III (AT&T)
 network:
diff --git a/_data/devices/d2spr.yml b/_data/devices/d2spr.yml
index 7cc27a6..7760d58 100644
--- a/_data/devices/d2spr.yml
+++ b/_data/devices/d2spr.yml
@@ -19,7 +19,7 @@ height:
 image: i9300.png
 install_method: heimdall
 kernel: android_kernel_samsung_d2
-maintainers:
+maintainers: []
 models: [SPH-L710]
 name: Galaxy S III (Sprint)
 network:
diff --git a/_data/devices/d2tmo.yml b/_data/devices/d2tmo.yml
index 1f42a6e..f15085d 100644
--- a/_data/devices/d2tmo.yml
+++ b/_data/devices/d2tmo.yml
@@ -19,7 +19,7 @@ height:
 image: i9300.png
 install_method: heimdall
 kernel: android_kernel_samsung_d2
-maintainers:
+maintainers: []
 models: [SGH-T999]
 name: Galaxy S III (T-Mobile)
 network:
diff --git a/_data/devices/d2vzw.yml b/_data/devices/d2vzw.yml
index fa85ba5..7b84a60 100644
--- a/_data/devices/d2vzw.yml
+++ b/_data/devices/d2vzw.yml
@@ -19,7 +19,7 @@ height:
 image: i9300.png
 install_method: heimdall
 kernel: android_kernel_samsung_d2
-maintainers:
+maintainers: []
 models: [SCH-I535]
 name: Galaxy S III (Verizon)
 network:
diff --git a/_data/devices/ghost.yml b/_data/devices/ghost.yml
index 63db2e0..9e95230 100644
--- a/_data/devices/ghost.yml
+++ b/_data/devices/ghost.yml
@@ -19,7 +19,7 @@ height: 129.3 mm (5.09 in)
 image: ghost.png
 install_method: fastboot_motorola
 kernel: android_kernel_motorola_ghost
-maintainers:
+maintainers: []
 models: [xt1052, xt1053, xt1055, xt1056, xt1058, xt1060]
 name: Moto X
 network:
diff --git a/_data/devices/gts210ltexx.yml b/_data/devices/gts210ltexx.yml
index 1396f5a..663bdf2 100644
--- a/_data/devices/gts210ltexx.yml
+++ b/_data/devices/gts210ltexx.yml
@@ -10,7 +10,7 @@ codename: gts210ltexx
 cpu: Cortex A57/A53
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz, 4 x 1.3 GHz
-current_branch: [14.1]
+current_branch: 14.1
 depth: 5.6 mm (0.22 in)
 download_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Down</kbd> + <kbd>Power</kbd>.
 gpu: Mali-T760 MP6
diff --git a/_data/devices/gts210wifi.yml b/_data/devices/gts210wifi.yml
index 0823a4b..58fe553 100644
--- a/_data/devices/gts210wifi.yml
+++ b/_data/devices/gts210wifi.yml
@@ -10,7 +10,7 @@ codename: gts210wifi
 cpu: Cortex A57/A53
 cpu_cores: '8'
 cpu_freq: 4 x 1.9 GHz, 4 x 1.3 GHz
-current_branch: [14.1]
+current_branch: 14.1
 depth: 5.6 mm (0.22 in)
 download_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Down</kbd> + <kbd>Power</kbd>.
 gpu: Mali-T760 MP6
diff --git a/_data/devices/ham.yml b/_data/devices/ham.yml
index c5d5f3c..7c5705c 100644
--- a/_data/devices/ham.yml
+++ b/_data/devices/ham.yml
@@ -10,7 +10,7 @@ codename: ham
 cpu: Krait 400
 cpu_cores: '4'
 cpu_freq: 2.5 GHz
-current_branch: [14.1]
+current_branch: 14.1
 depth: 8.9 mm (0.35 in)
 download_boot: With the device powered off, hold <kbd>Volume Down</kbd> + <kbd>Volume Up</kbd> and connect the USB cable.
 gpu: Adreno 330
diff --git a/_data/devices/i9305.yml b/_data/devices/i9305.yml
index 630ba8d..a767699 100644
--- a/_data/devices/i9305.yml
+++ b/_data/devices/i9305.yml
@@ -10,7 +10,7 @@ codename: i9305
 cpu: Cortex A9
 cpu_cores: '4'
 cpu_freq: 1.4 GHz
-current_branch: [14.1]
+current_branch: 14.1
 depth: 8.6 mm (0.339 in)
 download_boot: With the device powered off, hold <kbd>Home</kbd> + <kbd>Volume Down</kbd> + <kbd>Power</kbd>.
 gpu: Mali-400
@@ -19,7 +19,7 @@ height: 137 mm (5.4 in)
 image: i9300.png
 install_method: heimdall
 kernel: android_kernel_samsung_smdk4412
-maintainers:
+maintainers: []
 models: [GT-I9305, GT-I9305T, GT-I9305N]
 name: Galaxy S III (LTE / International)
 network:
diff --git a/_data/devices/libra.yml b/_data/devices/libra.yml
index d343f8c..e6dd695 100644
--- a/_data/devices/libra.yml
+++ b/_data/devices/libra.yml
@@ -31,7 +31,6 @@ ram: 2/3 GB
 recovery_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Power</kbd>.
   Keep holding both buttons until the "MI" logo appears on the screen, then release.
 release: 2015-09
-required_bootloader:
 screen: 127 mm (5.0 in)
 screen_ppi: 441
 screen_res: 1920x1080
diff --git a/_data/devices/m7.yml b/_data/devices/m7.yml
index 3543df0..8667fb8 100644
--- a/_data/devices/m7.yml
+++ b/_data/devices/m7.yml
@@ -19,7 +19,7 @@ height: 137.4 mm
 image: m7.png
 install_method: fastboot_htc
 kernel: android_kernel_htc_msm8960
-maintainers:
+maintainers: []
 name: One (GSM)
 network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
diff --git a/_data/devices/m7vzw.yml b/_data/devices/m7vzw.yml
index c3503fd..d95ff3c 100644
--- a/_data/devices/m7vzw.yml
+++ b/_data/devices/m7vzw.yml
@@ -19,7 +19,7 @@ height: 137.4 mm
 image: m7.png
 install_method: fastboot_htc
 kernel: android_kernel_htc_msm8960
-maintainers:
+maintainers: []
 name: One (Verizon)
 network:
 - {tech: 2G, bands: '800 1900 MHz CDMA' }
diff --git a/_data/devices/maguro.yml b/_data/devices/maguro.yml
index 72aefc1..1cf2b53 100644
--- a/_data/devices/maguro.yml
+++ b/_data/devices/maguro.yml
@@ -13,7 +13,7 @@ cpu_freq: 1.2 GHz
 current_branch: 13.0
 depth: 8.9 mm
 download_boot: With the device powered off, hold <kbd>Volume Down</kbd> + <kbd>Volume Up</kbd> + <kbd>Power</kbd> until the bootloader appears, then release the buttons.
-gpu: <kbd>Power</kbd>VR SGX540 @ 384 Mhz
+gpu: PowerVR SGX540 @ 384 Mhz
 has_recovery_partition: true
 height: 135.5 mm
 image: tuna.png
diff --git a/_data/devices/mt2.yml b/_data/devices/mt2.yml
index 63cca35..e50cff4 100644
--- a/_data/devices/mt2.yml
+++ b/_data/devices/mt2.yml
@@ -15,9 +15,9 @@ download_boot: With the device powered off, hold <kbd>Volume Down</kbd> + <kbd>P
 gpu: Adreno 305
 has_recovery_partition: true
 image: mt2.png
-install_method:
+install_method: fastboot_huawei
 kernel: android_kernel_huawei_msm8928
-maintainers:
+maintainers: []
 name: Ascend Mate 2 4G
 network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
diff --git a/_data/devices/n5100.yml b/_data/devices/n5100.yml
index a73c165..24c6d78 100644
--- a/_data/devices/n5100.yml
+++ b/_data/devices/n5100.yml
@@ -10,7 +10,7 @@ codename: n5100
 cpu: Cortex A9
 cpu_cores: '4'
 cpu_freq: 1.6 GHz
-current_branch: [14.1]
+current_branch: 14.1
 depth: 8.0 mm (0.31 in)
 download_boot: With the device powered off, hold <kbd>Volume Down</kbd> + <kbd>Power</kbd>.
 gpu: Mali-400
diff --git a/_data/devices/n5110.yml b/_data/devices/n5110.yml
index dba1114..50f5407 100644
--- a/_data/devices/n5110.yml
+++ b/_data/devices/n5110.yml
@@ -10,7 +10,7 @@ codename: n5110
 cpu: Cortex A9
 cpu_cores: '4'
 cpu_freq: 1.6 GHz
-current_branch: [14.1]
+current_branch: 14.1
 depth: 8.0 mm (0.31 in)
 download_boot: With the device powered off, hold <kbd>Volume Down</kbd> + <kbd>Power</kbd>.
 gpu: Mali-400
diff --git a/_data/devices/n5120.yml b/_data/devices/n5120.yml
index 795b98b..483c2c5 100644
--- a/_data/devices/n5120.yml
+++ b/_data/devices/n5120.yml
@@ -10,7 +10,7 @@ codename: n5120
 cpu: Cortex A9
 cpu_cores: '4'
 cpu_freq: 1.6 GHz
-current_branch: [14.1]
+current_branch: 14.1
 depth: 8.0 mm (0.31 in)
 download_boot: With the device powered off, hold <kbd>Volume Down</kbd> + <kbd>Power</kbd>.
 gpu: Mali-400
diff --git a/_data/devices/nicki.yml b/_data/devices/nicki.yml
index a20d8d6..81d3ab0 100644
--- a/_data/devices/nicki.yml
+++ b/_data/devices/nicki.yml
@@ -17,7 +17,7 @@ has_recovery_partition: true
 image: nicki.png
 install_method: fastboot_sony
 kernel: android_kernel_sony_msm8x27
-maintainers:
+maintainers: []
 name: Xperia M
 network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
diff --git a/_data/devices/peach.yml b/_data/devices/peach.yml
index a0624f5..296de93 100644
--- a/_data/devices/peach.yml
+++ b/_data/devices/peach.yml
@@ -19,7 +19,7 @@ height: 145.5 mm (5.72 in)
 image: peach.png
 install_method: fastboot_nexus
 kernel: android_kernel_ark_msm8916
-maintainers:
+maintainers: []
 name: Benefit A3
 network:
 - {tech: 2G, bands: '850, 900, 1800, 1900' }
diff --git a/_data/devices/t6.yml b/_data/devices/t6.yml
index eb4fd2a..11ac79b 100644
--- a/_data/devices/t6.yml
+++ b/_data/devices/t6.yml
@@ -20,7 +20,7 @@ height: 164.5 mm (6.48 in)
 image: t6.png
 install_method: fastboot_htc
 kernel: android_kernel_htc_msm8960
-maintainers:
+maintainers: []
 name: One Max (GSM)
 network:
 - {tech: 2G, bands: '850 900 1800 1900 MHz GSM' }
diff --git a/_data/devices/t6vzw.yml b/_data/devices/t6vzw.yml
index 9253467..04618e5 100644
--- a/_data/devices/t6vzw.yml
+++ b/_data/devices/t6vzw.yml
@@ -20,7 +20,7 @@ height: 164.5 mm (6.48 in)
 image: t6.png
 install_method: fastboot_htc
 kernel: android_kernel_htc_msm8960
-maintainers:
+maintainers: []
 name: One Max (Verizon)
 network:
 - {tech: 2G, bands: '800 1900 MHz CDMA' }
diff --git a/_data/devices/targa.yml b/_data/devices/targa.yml
index 2499ac5..f415fd4 100644
--- a/_data/devices/targa.yml
+++ b/_data/devices/targa.yml
@@ -12,7 +12,7 @@ cpu_cores: '2'
 cpu_freq: 1 GHz
 current_branch: 14.1
 download_boot: With the device powered off, hold <kbd>Volume Up</kbd> + <kbd>Volume Down</kbd> + <kbd>Power</kbd>.
-gpu: <kbd>Power</kbd>VR SGX540
+gpu: PowerVR SGX540
 has_recovery_partition: true
 image: targa.png
 install_method: fastboot_motorola
diff --git a/_data/devices/wt88047.yml b/_data/devices/wt88047.yml
index 69cffb9..f17e5c5 100644
--- a/_data/devices/wt88047.yml
+++ b/_data/devices/wt88047.yml
@@ -20,7 +20,7 @@ height: 134 mm (5.28 in)
 image: wt88047.png
 install_method: fastboot_xiaomi
 kernel: android_kernel_wingtech_msm8916
-maintainers:
+maintainers: []
 name: Redmi 2
 network:
 - {tech: 2G, bands: '900 1800 1900 MHz GSM' }
diff --git a/device_sample/sample.yml b/device_sample/sample.yml
index 6cd58bc..5e2977b 100644
--- a/device_sample/sample.yml
+++ b/device_sample/sample.yml
@@ -46,6 +46,6 @@ tree: android_device_lge_sample
 type: phone
 vendor: LG
 vendor_short: lge
-versions: [14.1, 15, 9000.4]
+versions: [13.0, 14.1, 15.1]
 width: 69.2 mm (2.72 in)
 wifi: 802.11 a/b/g/n/ac
diff --git a/test/schema-06.yml b/test/schema-06.yml
index f9ce971..a27a904 100644
--- a/test/schema-06.yml
+++ b/test/schema-06.yml
@@ -13,6 +13,13 @@ definitions:
     type:
     - integer
     - string
+  valid_branches:
+    description: Valid branches a device can be on
+    enum:
+    - 13.0
+    - 14.1
+    - 15.1
+    type: number
 properties:
   architecture:
     "$id": "/properties/architecture"
@@ -67,6 +74,58 @@ properties:
     type: array
   codename:
     "$id": "/properties/codename"
+    minLength: 1
+    type: string
+  cpu_cores:
+    "$id": "/properties/cpu_cores"
+    minimum: 1
+    minLength: 1
+    type:
+    - string
+    - integer
+  cpu_freq:
+    "$id": "/properties/cpu_freq"
+    minLength: 3
+    type: string
+  current_branch:
+    "$id": "/properties/current_branch"
+    "$ref": "#/definitions/valid_branches"
+  custom_unlock_cmd:
+    "$id": "/properties/custom_unlock_cmd"
+    minLength: 10
+    type: string
+  install_method:
+    "$id": "/properties/install_method"
+    enum:
+    - dd
+    - fastboot_custom
+    - fastboot_generic
+    - fastboot_huawei
+    - fastboot_htc
+    - fastboot_lg
+    - fastboot_motorola
+    - fastboot_nexus
+    - fastboot_oppo
+    - fastboot_sony
+    - fastboot_unlocked
+    - fastboot_xiaomi
+    - fastboot_zenfone
+    - heimdall
+  image:
+    "$id": "/properties/image"
+    pattern: ^\w*\.(png|jpg)$
+    type: string
+  kernel:
+    "$id": "/properties/kernel"
+    pattern: ^(android_kernel(_\w*){2}(-common)?|lge-kernel-\w*)$
+    type: string
+  maintainers:
+    "$id": "/properties/maintainers"
+    minItems: 0
+    type: array
+  name:
+    "$id": "/properties/name"
+    minLength: 1
     type: string
   release:
     "$id": "/properties/release"
@@ -83,6 +142,10 @@ properties:
         description: Map of model number to release date
         type: object
       type: array
+  required_bootloader:
+    "$id": "/properties/required_bootloader"
+    minItems: 1
+    type: array
   type:
     "$id": "/properties/type"
     enum:
@@ -147,14 +210,30 @@ properties:
     - yu
     - zte
     - zuk
+  versions:
+    "$id": "/properties/versions"
+    items:
+      "$id": "/properties/versions/items"
+      "$ref": "#/definitions/valid_branches"
+    minItems: 1
+    type: array
 required:
 - architecture
 - bluetooth
 - channels
 - codename
+- cpu_cores
+- cpu_freq
+- current_branch
+- install_method
+- image
+- kernel
+- maintainers
+- name
 - release
 - type
 - vendor
 - vendor_short
+- versions
 title: device
 type: object
-- 
2.17.1

